### ISSUE TITLE
Use ssh for cloning repositories

### ISSUE THREAD

**Issue by mzander** [TabbyML/tabby#2418]
In our self hosted gitlab we disabled cloning using https. It only allows ssh

I think it should be possible to switch the protocol within the repository settings.

---
Please reply with a 👍 if you want this feature.

**Comment by wsxiaoys** [comment]
cc @boxbeam does it make sense to use ssh based clone by default？

**Comment by boxbeam** [comment]
I don't think we can use it by default, since it requires a different mode of authentication. Cloning via ssh would need to be its own feature

**Comment by wsxiaoys** [comment]
Good point - i guess ssh based solution has to go with some pub key for deployment purpose.

**Comment by vladfaust** [comment]
In the [docs](https://tabby.tabbyml.com/docs/administration/context/#adding-a-repository-through-configuration-file), ssh is listed as an option for a repository.
However, specifying one doesn't work: no jobs are created, nothing.
Is it possible to use SSH for private Git projects currently?
